telemetry/src/metrics.rs

```diff
@@ -0,0 +1,174 @@
+//! [`Metrics`] and [`Status`]-related logic and functions.
+use std::{
+    sync::Arc,
+    time::{Duration, SystemTime},
+};
+
+use prometheus::{
+    core::{AtomicU64, GenericGauge, GenericGaugeVec},
+    Encoder, Histogram, HistogramOpts, HistogramVec, IntCounter, IntCounterVec, Opts, Registry,
+};
+use serde::{Deserialize, Serialize};
+
+/// Thin wrapper around duration that `impl`s [`Default`]
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct Uptime(Duration);
+
+impl Default for Uptime {
+    fn default() -> Self {
+        Self(Duration::from_millis(0))
+    }
+}
+
+/// Response body for GET status request
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize)]
+pub struct Status {
```

Comment:
is `Status` part of metrics? it's in the `metrics` module

Reply:
looking at it, it could be considered a part of metrics

Reply:
I think so. The only difference between `Status` and `prometheus` endpoints is the API, and the type of returned values.